### PR TITLE
[DOCS] Adds AUC ROC classification metric to the API examples

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a classification evaluation example API call that specifies `auc_roc` as an evaluation metric. It also adds an example response and footnotes to the snippets.

### Preview

[Classification examples](https://elasticsearch_63563.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/evaluate-dfanalytics.html#ml-evaluate-classification-example)

### CODE REVIEW

**Comment by szabosteve** [comment]
@elasticmachine test this please.

File: docs/reference/ml/df-analytics/apis/evaluate-dfanalytics.asciidoc [hunk 1]
```diff
@@ -497,3 +497,55 @@ predictions associated with the class.
...
+
+The API returns the following result:
+
+[source,console-result]
+--------------------------------------------------
+{
+  "classification" : {
+    "auc_roc" : {
+      "score" : 0.8941788639536681, <1>
+      "doc_count" : 4169 <2>
```
Comment (przemekwitek) [inline docs/reference/ml/df-analytics/apis/evaluate-dfanalytics.asciidoc:541]: With https://github.com/elastic/elasticsearch/pull/63302, `doc_count` is no longer reported.
Please remove.

Comment (szabosteve) [reply docs/reference/ml/df-analytics/apis/evaluate-dfanalytics.asciidoc]: @przemekwitek Done.

File: docs/reference/ml/df-analytics/apis/evaluate-dfanalytics.asciidoc [hunk 2]
```diff
@@ -497,3 +497,49 @@ predictions associated with the class.
...
+
+[source,console]
+--------------------------------------------------
+POST _ml/data_frame/_evaluate
+{
+   "index": "animal_classification",
+   "evaluation": {
+      "classification": { <1>
+         "actual_field": "animal_class", <2>
+         "predicted_field": "ml.animal_class_prediction", <3>
```
Comment (przemekwitek) [inline docs/reference/ml/df-analytics/apis/evaluate-dfanalytics.asciidoc:511]: This field is not used by `auc_roc` so you can remove it (but please double-check that the example is still working, there may be bugs...).

Comment (szabosteve) [reply docs/reference/ml/df-analytics/apis/evaluate-dfanalytics.asciidoc]: I double-checked, it works fine without `predicted_field`. I removed this line from the call.

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**